rust_src/src/frames.rs

```diff
@@ -414,4 +422,86 @@ pub fn delete_frame_lisp(frame: LispObject, force: bool) {
     }
 }
 
+/// Return the next frame in the frame list after FRAME.
+/// It considers only frames on the same terminal as FRAME.
+/// By default, skip minibuffer-only frames.
+/// If omitted, FRAME defaults to the selected frame.
+/// If optional argument MINIFRAME is nil, exclude minibuffer-only frames.
+/// If MINIFRAME is a window, include only its own frame
+/// and any frame now using that window as the minibuffer.
+/// If MINIFRAME is `visible', include all visible frames.
+/// If MINIFRAME is 0, include all visible and iconified frames.
+/// Otherwise, include all frames.
+#[lisp_fn(min = "0")]
+pub fn next_frame(frame: LispFrameOrSelected, miniframe: LispObject) -> LispFrameRef {
+    let frame_ref = frame.live_or_error();
+    let frame_obj = frame_ref.into();
+
+    // Track how many times have we passed FRAME in the list.
+    let mut passed = 0;
+
+    // Go through the list at most twice. This treats the frame list
+    // as a circular list allowing for 'next' to occur before FRAME.
+    // Once the desired frame is found in the list, the next frame that is
+    // a valid candidate will be returned regardless of its position.
+    while passed < 2 {
+        for_each_frame!(f => {
+	    if passed > 0 {
+	        let tmp = unsafe { candidate_frame(f.into(), frame_obj, miniframe) };
+	        if !tmp.is_nil() {
```

Comment:
I don't want to be pedantic and I would also prefer this way. But the team decided differently and I think we should try to use `is_not_nil` consistently.